Mail/Views/Bottom sheets/DiscoveryView.swift

```diff
@@ -0,0 +1,170 @@
+
+ This program is free software: you can redistribute it and/or modify
+ it under the terms of the GNU General Public License as published by
+ the Free Software Foundation, either version 3 of the License, or
+ (at your option) any later version.
+
+ This program is distributed in the hope that it will be useful,
+ but WITHOUT ANY WARRANTY; without even the implied warranty of
+ MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ GNU General Public License for more details.
+
+ You should have received a copy of the GNU General Public License
+ along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import InfomaniakCoreUI
+import InfomaniakDI
+import MailCore
+import MailResources
+import SwiftUI
+
+public struct DiscoveryItem {
+    public let image: MailResourcesImages
+    public let title: String
+    public let description: String
+    public let primaryButtonLabel: String
+    public let matomoCategory: MatomoUtils.EventCategory
+}
+
+public extension DiscoveryItem {
+    static let aiDiscovery = DiscoveryItem(
+        image: MailResourcesAsset.aiIllustration,
+        title: MailResourcesStrings.Localizable.aiDiscoveryTitle,
+        description: MailResourcesStrings.Localizable.aiDiscoveryDescription,
+        primaryButtonLabel: MailResourcesStrings.Localizable.buttonTry,
+        matomoCategory: .aiWriter
+    )
+
+    static let syncDiscovery = DiscoveryItem(
+        image: MailResourcesAsset.syncIllustration,
+        title: MailResourcesStrings.Localizable.syncCalendarsAndContactsTitle,
+        description: MailResourcesStrings.Localizable.syncCalendarsAndContactsDescriptioniOS,
+        primaryButtonLabel: MailResourcesStrings.Localizable.buttonStart,
+        matomoCategory: .syncAutoConfig
+    )
+
+    static let updateDiscovery = DiscoveryItem(
+        image: MailResourcesAsset.logoMailWithStar,
+        title: MailResourcesStrings.Localizable.updateAvailableTitle,
+        description: MailResourcesStrings.Localizable.updateAvailableDescription,
+        primaryButtonLabel: MailResourcesStrings.Localizable.buttonUpdate,
+        matomoCategory: .appUpdate
+    )
+}
+
+struct DiscoveryView: View {
+    @LazyInjectService private var matomo: MatomoUtils
+
+    @Environment(\.isCompactWindow) private var isCompactWindow
+    @Environment(\.dismiss) private var dismiss
+
+    let item: DiscoveryItem
+
+    @State private var willDiscoverNewFeature = false
+
+    let shouldPresentFeature: (Bool) -> Void
+    let completionHandler: (Bool) -> Void
+
+    var body: some View {
+        Group {
+            if isCompactWindow {
+                DiscoveryBottomSheetView(item: item, nowButton: didTouchNowButton, laterButton: didTouchLaterButton)
+            } else {
+                DiscoveryAlertView(item: item, nowButton: didTouchNowButton, laterButton: didTouchLaterButton)
+            }
+        }
+        .onAppear {
+            shouldPresentFeature(false)
```

Comment:
What is the use of the parameter of this function for you? Because we always call it with `false`?